ext/pdo/pdo_dbh.c

```diff
@@ -955,8 +956,15 @@ PHP_METHOD(PDO, lastInsertId)
 		Z_PARAM_STRING_OR_NULL(name, namelen)
 	ZEND_PARSE_PARAMETERS_END();
 
-	PDO_DBH_CLEAR_ERR();
 	PDO_CONSTRUCT_CHECK;
+
+	if (name && namelen == 0) {
```

Comment:
Where was this previously enforced?

Reply:
It was not, should I revert this?

Reply:
I think it's better to leave it alone. E.g. from a quick look, this parameter is completely ignored for mysql.